src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2054,7 +2073,14 @@ protected function getMorphs($name, $type, $id)
      */
     public function getMorphClass()
     {
-        return $this->morphClass ?: get_class($this);
+        $morphMap = Relation::morphMap();
+        $class = get_class($this);
+
+        if (! empty($morphMap) && in_array($class, $morphMap)) {
+            return array_flip($morphMap)[$class];
```

Comment:
performance of this function is surely lower than if you were to use array_search.